doc/using_mxnet.rst

```diff
@@ -1,822 +1,784 @@
+
+To provide your own save function, define a ``save`` function in your training script:
+
+.. code:: python
+
+    def save(model, model_dir):
+        pass
+
+The function should take two arguments:
+
+-  ``model``: This is the object that was returned from your ``train``
+   function. If your ``train`` function does not return an object, it
+   will be ``None``. You are free to return an object of any type from
+   ``train``, you do not have to return ``Module`` or ``Gluon`` API
+   specific objects.
+-  ``model_dir``: This is the string path on the SageMaker training host
+   where you save your model. Files created in this directory will be
+   accessible in S3 after your SageMaker Training Job completes.
+
+After your ``train`` function completes, SageMaker will invoke ``save`` with the object returned from ``train``.
+
+**Note: How to save Gluon models with SageMaker**
+
+If your train function returns a Gluon API ``net`` object as its model, you'll need to write your own ``save`` function. You will want to serialize the ``net`` parameters. Saving ``net`` parameters is covered in the `Serialization section <http://gluon.mxnet.io/chapter03_deep-neural-networks/serialization.html>`__ of the collaborative Gluon deep-learning book `"The Straight Dope" <http://gluon.mxnet.io/index.html>`__.
+
+Updating your MXNet training script
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The structure for training scripts changed with MXNet version 1.3.
+The ``train`` function is no longer be required; instead the training script must be able to be run as a standalone script.
+In this way, the training script is similar to a training script you might run outside of SageMaker.
+
+There are a few steps needed to make a training script with the old format compatible with the new format.
+
+First, add a `main guard <https://docs.python.org/3/library/__main__.html>`__ (``if __name__ == '__main__':``).
+The code executed from your main guard needs to:
+
+1. Set hyperparameters and directory locations
+2. Initiate training
+3. Save the model
+
+Hyperparameters will be passed as command-line arguments to your training script.
+In addition, the container will define the locations of input data and where to save the model artifacts and output data as environment variables rather than passing that information as arguments to the ``train`` function.
+You can find the full list of available environment variables in the `SageMaker Containers README <https://github.com/aws/sagemaker-containers#list-of-provided-environment-variables-by-sagemaker-containers>`__.
+
+We recommend using `an argument parser <https://docs.python.org/3.5/howto/argparse.html>`__ for this part.
+Using the ``argparse`` library as an example, the code would look something like this:
+
+.. code:: python
+
+    import argparse
+    import os
+
+    if __name__ == '__main__':
+        parser = argparse.ArgumentParser()
+
+        # hyperparameters sent by the client are passed as command-line argume
```

Comment:
Does the requirement.txt thing still work?

Reply:
yes. it's disabled only for TF script mode.